includes/amp-rest-functions.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * AMP Rest Functions
+ *
+ * @package AMP
+ */
+
+/**
+ * Register the comments REST endpoint.
+ *
+ * @since 0.7
+ */
+function amp_register_endpoints() {
+
+	register_rest_route( 'amp/v1', '/comments/(?P<id>\d+)', array(
+		'methods'  => 'GET',
+		'callback' => 'amp_get_comments',
+	) );
+
+}
+
+/**
+ * Get comments for the Post and Parent
+ *
+ * @since 0.7
+ * @param WP_REST_Request $request The REST request.
+ *
+ * @return array The array of comments.
+ */
+function amp_get_comments( $request ) {
+
+	$id     = $request->get_param( 'id' );
+	$return = array(
+		'items' => array(
+			'comment' => amp_get_comments_recursive( $id, 0 ),
+		),
+	);
+
+	return $return;
+}
+
+/**
+ * Recursively get comments for the Post and Parent
+ *
+ * @since 0.7
+ * @param int $id The post ID.
+ * @param int $parent The comment parent.
+ *
+ * @return array The array of comments.
+ */
+function amp_get_comments_recursive( $id, $parent ) {
+
+	$comments = get_comments( array(
+		'post_ID' => $id,
+		'parent'  => $parent,
+		'order'   => 'ASC',
+	) );
+
+	$return = array();
+	foreach ( $comments as $comment ) {
+		$GLOBALS['comment']    = $comment; // WPCS: override ok.
+		$comment->comment_date = get_comment_date();
+		$comment->comment_time = get_comment_time();
+		$comment->comment      = amp_get_comments_recursive( $id, $comment->comment_ID );
```

Comment:
I believe this will have performance problems, as it could result in a lot of DB queries. If you look at `wp_list_comments()` I believe you can see it gets a flat list of all comments for the post, and then it passes it into `Walker::paged_walk()` which then goes about identifying which comments are children of other comments.

Reply:
That is, `AMP_Comment_Walker:: paged_walk()` here.

Reply:
@westonruter Agreed. Can revise it to be more efficient.

Reply:
@westonruter how about this? b7c5963